components/reference-implementation-openbanking-consumerdatastandards-au/src/main/java/org/wso2/openbanking/consumerdatastandards/au/extensions/utils/CDSConsentAuthPersistUtil.java

```diff
@@ -0,0 +1,313 @@
+import org.wso2.openbanking.consumerdatastandards.au.extensions.constants.CommonConstants;
+import org.wso2.openbanking.consumerdatastandards.au.extensions.exceptions.CDSConsentException;
+import org.wso2.openbanking.consumerdatastandards.au.extensions.gen.model.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class for CDS Consent Auth persistence operations.
+ */
+public class CDSConsentAuthPersistUtil {
+
+    private static final Log log = LogFactory.getLog(CDSConsentAuthPersistUtil.class);
+
+    /**
+     * Method to handle the persist consent request.
+     * @param persistAuthorizedConsentRequestBody
+     * @return Detailed Consent Resource
+     */
+    public static SuccessResponsePersistAuthorizedConsentData CDSConsentPersist(
+            PersistAuthorizedConsentRequestBody persistAuthorizedConsentRequestBody) throws CDSConsentException {
+
+        String consentStatus;
+        String authStatus;
+        long validityTime;
+
+        SuccessResponsePersistAuthorizedConsentData persistAuthorizedConsentDataRes =
+                new SuccessResponsePersistAuthorizedConsentData();
+
+        try {
+            PersistAuthorizedConsent requestData = persistAuthorizedConsentRequestBody.getData();
+
+            DetailedConsentResourceDataWithAmendments updatedConsent = new DetailedConsentResourceDataWithAmendments();
+
+            UserGrantedData userGrantedData = requestData.getUserGrantedData();
+
+            // Convert user granted data to JSON object
+            JSONObject consumerInputData = CommonConsentExtensionUtils.convertObjectToJson(userGrantedData);
+
+            // Get authorisedResources from user granted data
+            AuthorizedResources authorizedResources = userGrantedData.getAuthorizedResources();
+
+            // Get authorized data list from authorized resources
+            List<AuthorizedResourcesAuthorizedDataInner> authorizedDataInners = authorizedResources.getAuthorizedData();
+
+            // Get metadata object from authorized resources
+            Object metadataObject = authorizedResources.getMetadata();
+
+            //Set Consent Type
+            String consentType = authorizedResources.getType();
+
+            //Check whether the consent is approved or not and assign consent status.
+            boolean isApproved = requestData.getIsApproved();
+            if (!isApproved) {
+                consentStatus = CommonConstants.REJECTED_STATUS;
+                authStatus = CommonConstants.REJECTED_STATUS;
+            } else {
+                consentStatus = CommonConstants.AUTHORIZED_STATUS;
+                authStatus = CommonConstants.AUTHORIZED_STATUS;
+            }
+
+            //Check if account ids are available in the request when consent is approved.
+            if (isApproved) {
```

Comment:
Can't we add this inside the above else block?

Reply:
Fixed in [6072521](https://github.com/wso2/reference-implementation-consumerdatastandards-au/pull/132/commits/6072521179a92335986ad72ca75eed788dbb8d64)